Multiple minor pipeline runtime updates

lresende opened the pull request:
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

lresende commented on elyra/pipeline/tests/test_pipeline_parser.py:
```diff
@@ -27,7 +27,11 @@ def valid_operation():
+                     image='{{image}}',
+                     vars=["var1=var1", "var2=var2"],
+                     file_dependencies=["a.txt", "b.txt", "c.txt"],
+                     outputs=["d.txt", "e.txt", "f.txt"],
```
Yes, I wasn't sure if I should do this here (which is when i found the missing field) or work with @akchinSTC to make it part of his changes. Maybe I should make mine WIP and wait.

kevin-bates replied:
If we'd be mixing name updates across two different PRs, then, yes, we should coordinate this and probably wait for @akchinSTC's changes or collaborate on a shared branch.

lresende commented on packages/pipeline-editor/src/properties.json:
```diff
@@ -35,40 +35,40 @@
         },
         "place_holder_text": {
-          "default": "One filename per line"
+          "default": "Local file dependencies that needs to be copied to remote execution environment\nOne filename or extension wildcard (eg. *.py) per line"
```
Thanks

lresende commented on packages/pipeline-editor/src/properties.json:
```diff
@@ -35,40 +35,40 @@
       },
       {
-        "parameter_ref": "vars",
+        "parameter_ref": "recursive_dependencies",
```
Yes, coming on #544

lresende commented:
@ajbozarth Here is the visual for the updated properties


![image](https://user-images.githubusercontent.com/382917/83073201-e2a5b980-a024-11ea-9720-7ffd20510f63.png)

kevin-bates commented:
Thanks Luciano. Should these sentences include periods?  (Not sure what the UX book says here.)
Also, can we expand `var` to `variable` in the second line of the Environment Variables?
I see that some headers use capitalization of each word (last two), others do not (first two).  I think we need to be consistent here - either way.

ajbozarth commented:
I agree with @kevin-bates we should spell out `variable` in the text, we should be consistent in title capitalization (I'm fine with either choice though) and since the placeholder text is now 2 lines we should include periods (I believe that is typical UX).

ajbozarth commented on packages/pipeline-editor/src/properties.json:
```diff
@@ -35,40 +35,40 @@
-        },
-        "place_holder_text": {
-          "default": "One var per line in the format ENV_VAR=value"
+          "default": "Include subdirectories in dependencies (may increase submission time)"
```
I agree this text should be clarified, but as we discussed in our earlier meeting it seems it's currently "broken" so perhaps we should update this text once we've decided on the behavior and fixed it. (So either `Include subdirectories in wildcard dependencies (may increase submission time)' or `Include subdirectories in directory dependencies (may increase submission time)`)

lresende commented:
Rebased and updated with review feedback

![image](https://user-images.githubusercontent.com/382917/83105208-19a0bd00-a06f-11ea-9558-7ca60d4e3cfe.png)
